[imgui] Set C++ 11 standard in CMakeLists.txt

****

**Is your feature request related to a problem? Please describe.**
When compiling imgui with vcpkg on Ubuntu Xenial the compiler rightfully complains about needing C++ 11 for `constexpr` which is used in imgui code.

**Proposed solution**
The CMakeLists.txt should set the compiler flags for C++ 11, as the code contains C++ 11 features.

**Describe alternatives you've considered**
I tried setting `CMAKE_CXX_STANDARD=11` from the outside (e.g. with `cmake -DCMAKE_CXX_STANDARD=11 ....`), which should enable C++ 11 in imgui code, but it does not get propagated by vcpkg. 
Switching to another compiler so it defaults to C++ 11 might be an last resort alternative.

**Additional context**
This is the compilation error that I get on a fresh ubuntu:xenial. Well mostly fresh: as vcpkg needs at least cmake-3.7 and ubuntu:xenial does not provide it, cmake-3.8.2 is compiled from source - but that should make no difference on the compilation error:

```
.../vcpkg/buildtrees/imgui/src/v1.88-34c93572a3.clean/imgui.h:252:5: error: ‘constexpr’ does not name a type
     constexpr ImVec2()                      : x(0.0f), y(0.0f) { }
     ^
.../vcpkg/buildtrees/imgui/src/v1.88-34c93572a3.clean/imgui.h:252:5: note: C++11 ‘constexpr’ only available with -std=c++11 or -std=gnu++11
.../vcpkg/buildtrees/imgui/src/v1.88-34c93572a3.clean/imgui.h:253:5: error: ‘constexpr’ does not name a type
     constexpr ImVec2(float _x, float _y)    : x(_x), y(_y) { }
     ^
.../vcpkg/buildtrees/imgui/src/v1.88-34c93572a3.clean/imgui.h:253:5: note: C++11 ‘constexpr’ only available with -std=c++11 or -std=gnu++11

... goes on forever...
```

--- a/ports/imgui/CMakeLists.txt
+++ b/ports/imgui/CMakeLists.txt
@@ -28,6 +28,8 @@ target_sources(
         ${CMAKE_CURRENT_SOURCE_DIR}/misc/cpp/imgui_stdlib.cpp
 )
 
+set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
+
 if(IMGUI_BUILD_ALLEGRO5_BINDING)
     find_path(ALLEGRO5_INCLUDE_DIRS allegro5/allegro.h)
     target_include_directories(${PROJECT_NAME} PRIVATE ${ALLEGRO5_INCLUDE_DIRS})

--- a/ports/imgui/CMakeLists.txt
+++ b/ports/imgui/CMakeLists.txt
@@ -28,7 +28,7 @@ target_sources(
         ${CMAKE_CURRENT_SOURCE_DIR}/misc/cpp/imgui_stdlib.cpp
 )
 
-set_property(TARGET ${PROJECT_NAME} PROPERTY CXX_STANDARD 11)
+target_compile_features(${PROJECT_NAME} PUBLIC cxx_std_11)
 
 if(IMGUI_BUILD_ALLEGRO5_BINDING)
     find_path(ALLEGRO5_INCLUDE_DIRS allegro5/allegro.h)

--- a/ports/imgui/vcpkg.json
+++ b/ports/imgui/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "imgui",
   "version": "1.88",
+  "port-version": 1,
   "description": "Bloat-free Immediate Mode Graphical User interface for C++ with minimal dependencies.",
   "homepage": "https://github.com/ocornut/imgui",
   "license": "MIT",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -12,7 +12,6 @@
       "baseline": "3.0.5",
       "port-version": 0
     },
-
     "abseil": {
       "baseline": "20211102.1",
       "port-version": 0
@@ -2963,7 +2962,7 @@
     },
     "imgui": {
       "baseline": "1.88",
-      "port-version": 0
+      "port-version": 1
     },
     "imgui-sfml": {
       "baseline": "2.5",

--- a/versions/i-/imgui.json
+++ b/versions/i-/imgui.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "2a4c55df4b895535fba9e3934ed64cef92bf4a4d",
+      "version": "1.88",
+      "port-version": 1
+    },
+    {
       "git-tree": "c28ebbdbe22a87ce01c3b2b6c15bed036721c6a0",
       "version": "1.88",
       "port-version": 0

